.travis.yml

```diff
@@ -29,6 +29,17 @@ install:
 script:
   # Run checks for non-GUI code
   - ./test.sh
+
+# TODO: Disabled because $TRAVIS_BRANCH sometimes can be set incorrectly
```

Comment:
This clutters the current file, could you just keep it in test.sh but add another TODO to extract this into a separate script, so that it only runs on Travis but not locally? .travis.yml then would invoke both test.sh and that script.

Reply:
Extracted all into a separate script.